driver/src/contracts/stablex_auction_element.rs

```diff
@@ -0,0 +1,313 @@
+        let account_id = H160::from(&bytes[0..20]);
+
+        // these go together (since sell_token_balance is emitted as u256 and treated as u128
+        let sell_token_balance = BigEndian::read_u128(&bytes[36..52]);
+        let sell_token_balance_padding = BigEndian::read_u128(&bytes[20..36]);
+        assert_eq!(
+            sell_token_balance_padding, 0,
+            "User has too large balance to handle."
+        );
+
+        let buy_token = u16::from_le_bytes([bytes[53], bytes[52]]);
+        let sell_token = u16::from_le_bytes([bytes[55], bytes[54]]);
+        let valid_from = U256::from(u32::from_le_bytes([
+            bytes[59], bytes[58], bytes[57], bytes[56],
+        ]));
+        let valid_until = U256::from(u32::from_le_bytes([
+            bytes[63], bytes[62], bytes[61], bytes[60],
+        ]));
+        let is_sell_order = bytes[64] > 0;
+        let numerator = BigEndian::read_u128(&bytes[65..81]);
+        let denominator = BigEndian::read_u128(&bytes[81..97]);
+        let amount = BigEndian::read_u128(&bytes[97..113]);
+        let (buy_amount, sell_amount) =
+            compute_buy_sell_amounts(numerator, denominator, amount, is_sell_order);
+        let order_counter = order_count.entry(account_id).or_insert(0);
+        *order_counter += 1;
+        StableXAuctionElement {
+            valid_from,
+            valid_until,
+            sell_token_balance,
+            order: Order {
+                batch_information: Some(BatchInformation {
+                    slot_index: *order_counter - 1,
+                    slot: U256::from(0),
+                }),
+                account_id,
+                buy_token,
+                sell_token,
+                buy_amount,
+                sell_amount,
+            },
+        }
+    }
+}
+
+fn compute_buy_sell_amounts(
+    numerator: u128,
+    denominator: u128,
+    remaining: u128,
+    is_sell_order: bool,
+) -> (u128, u128) {
+    // TODO - use u128_to_u256 for overflow handling (this is a conversion used in naive_solver)
+    assert!(
+        remaining <= denominator,
+        "Smart contract should never allow this inequality"
+    );
+    let mut other = 0 as u128;
+    let (buy_amount, sell_amount) = if is_sell_order {
+        // 0 on sellAmount (remaining <= denominator) is nonsense, but solver can handle it.
+        // 0 on buyAmount (numerator) is a Market Sell Order.
+        if denominator > 0 {
+            // up-casting to handle overflow
+            let top = u128_to_u256(remaining) * u128_to_u256(numerator);
+            let bottom = u128_to_u256(denominator);
+            // Recall that ceil(p / float(q)) == (p + q - 1) / q
+            let other_u256 = (top + bottom - 1) / bottom;
+
+            // Conversion never overflows because remaining <= denominator
+            let x = other_u256.0;
+            other = u128::from(x[0]) + u128::from(x[1]) * (2 as u128).pow(64);
```

Comment:
Here you could just use `u128::from_str(x.toString()).unwrap()` but I'd suggest creating a separate helper method next to `u128_to_u256`

Reply:
```
91 |             let x = u128::from_str(x.toString()).unwrap();
   |                           ^^^^^^^^ function or associated item not found in `u128`
```

Reply:
@fleupold  - do you think we should update our dependencies here. My PR to parity was closed because our dependencies are depreciated. 

https://github.com/paritytech/primitives/pull/68

Reply:
I think we should look into it. However, I'm not sure how easy it is since the graph also depends on web3 types and I'm not sure if the versions are compatible.

Reply:
Regarding the from_str. It is implemented (https://doc.rust-lang.org/std/primitive.u128.html#impl-FromStr)

Maybe you have to import the trait `use std::str::FromStr`

Reply:
Oh man! If that is really the case, I think we may recover from some other inconveniences in other places as well. I'll check this!

Reply:
@josojo  and I were really stumped by this the other day!

Reply:
All good now, that worked.